pypdf/_writer.py

```diff
@@ -497,7 +554,8 @@ def add_page(
         Returns:
             The added PageObject.
         """
-        return self._add_page(page, list.append, excluded_keys)
+        assert self.flattened_pages is not None
```

Comment:
Shouldn't this raise a proper error?

Reply:
flattened_pages is now properly initialized since the constructor. this is just for mypy